File: docs/vsto/codesnippet/VisualBasic/Trin_VstcoreTroubleshootingWordVB/ThisDocument.vb
'<Snippet2>
Imports Word = Microsoft.Office.Interop.Word
'</Snippet2>


Public Class ThisDocument

    Private Sub ThisDocument_Startup(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Startup
    End Sub

    Private Sub ThisDocument_Shutdown(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Shutdown
    End Sub


    Class Document
    End Class


    '--------------------------------------------------------------------------
    Sub Test2()

        '<Snippet4>
        Dim doc As Document  ' Class is ambiguous
        '</Snippet4>

        doc = New Document
    End Sub


    '--------------------------------------------------------------------------
    Sub Test1()

        '<Snippet3>
        Dim doc As Word.Document
        '</Snippet3>

        doc = New Word.Document
    End Sub


    '--------------------------------------------------------------------------
    '<Snippet1>
    Dim form1 As SampleForm

    Sub OpenForm()
        form1 = New SampleForm
        form1.Show()  ' Show the form modelessly.
    End Sub

    Sub ForceShutdown()

        ' Completely close the form if it is still running.
        ' Note that hiding the form might not work by itself.

        If (Not form1 Is Nothing) Then
            form1.Close()
            form1.Dispose()
            form1 = Nothing
        End If

        Me.Close()
    End Sub
    '</Snippet1>
End Class

File: docs/vsto/codesnippet/CSharp/Trin_VstcoreTroubleshootingWordCS/ThisDocument.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualStudio.Tools.Applications.Runtime;
//<Snippet2>
using Word = Microsoft.Office.Interop.Word;
//</Snippet2>
using Office=Microsoft.Office.Core;

namespace Trin_VstcoreTroubleshootingWordCS
{
    public partial class ThisDocument
    {
        private void ThisDocument_Startup(object sender, System.EventArgs e)
        {
        }

        private void ThisDocument_Shutdown(object sender, System.EventArgs e)
        {
        }

        private void InternalStartup()
        {
            this.Startup+=new System.EventHandler(ThisDocument_Startup);
            this.Shutdown+=new System.EventHandler(ThisDocument_Shutdown);
        }

        class Document{}


        //---------------------------------------------------------------------
        void Test2()
        {
            //<Snippet4>
            Document doc;  // Class is ambiguous
            //</Snippet4>

            doc = new Document();
        }


        //---------------------------------------------------------------------
        void Test1()
        {
            //<Snippet3>
            Word.Document doc;
            //</Snippet3>

            doc = new Microsoft.Office.Interop.Word.Document();
        }
        

        //---------------------------------------------------------------------
        //<Snippet1>
        SampleForm form1;

        private void OpenForm()
        {
            form1 = new SampleForm();
            form1.Show();  // Show form modelessly.
        }

        private void ForceShutdown()
        {
            // Completely close the form if it is still running.
            // Note that hiding the form might not work by itself.

            if (form1 != null)
            {
                form1.Close();
                form1.Dispose();
                form1 = null;
            }
            object saveChanges = Word.WdSaveOptions.wdSaveChanges; 
            this.Close(ref saveChanges, ref missing, ref missing);
        }
        //</Snippet1>
    }
}
